src/js/binaryen.js-post.js

```diff
@@ -4768,8 +4798,145 @@ Module['I31Get'] = makeExpressionWrapper({
+  ElementSegment.prototype['valueOf'] = function() {
+    return this[thisPtr];
+  };
+  return ElementSegment;
+})();
+
+// Export wrapper
+Module['Export'] = (() => {
+  // Closure compiler doesn't allow multiple `Export`s at top-level, so:
+  function Export(func) {
+    if (!(this instanceof Export)) {
+      if (!func) return null;
+      return new Export(func);
+    }
+    if (!func) throw Error("function reference must not be null");
+    this[thisPtr] = func;
+  }
+  Export['getKind'] = function(export_) {
+    return Module['_BinaryenExportGetKind'](export_);
+  };
+  Export['getName'] = function(export_) {
+    return UTF8ToString(Module['_BinaryenExportGetName'](export_));
+  };
+  Export['getValue'] = function(export_) {
+    return UTF8ToString(Module['_BinaryenExportGetValue'](export_));
+  };
+  deriveWrapperInstanceMembers(Export.prototype, Export);
+  Export.prototype['valueOf'] = function() {
+    return this[thisPtr];
+  };
+  return Export;
+})();
 
+// GlobalImport wrapper
+Module['GlobalImport'] = (() => {
+  // Closure compiler doesn't allow multiple `GlobalImport`s at top-level, so:
+  function GlobalImport(func) {
+    if (!(this instanceof GlobalImport)) {
+      if (!func) return null;
+      return new GlobalImport(func);
+    }
+    if (!func) throw Error("function reference must not be null");
+    this[thisPtr] = func;
+  }
+  GlobalImport['getModule'] = function(global) {
+    return UTF8ToString(Module['_BinaryenGlobalImportGetModule'](global));
+  };
+  GlobalImport['getBase'] = function(global) {
+    return UTF8ToString(Module['_BinaryenGlobalImportGetBase'](global));
+  };
+  deriveWrapperInstanceMembers(GlobalImport.prototype, GlobalImport);
+  GlobalImport.prototype['valueOf'] = function() {
+    return this[thisPtr];
+  };
+  return GlobalImport;
+})();
+
+// Global wrapper
+Module['Global'] = (() => {
+  // Closure compiler doesn't allow multiple `Global`s at top-level, so:
+  function Global(func) {
+    if (!(this instanceof Global)) {
+      if (!func) return null;
+      return new Global(func);
+    }
+    if (!func) throw Error("function reference must not be null");
+    this[thisPtr] = func;
+  }
+  Global['getName'] = function(global) {
+    return UTF8ToString(Module['_BinaryenGlobalGetName'](global));
+  };
+  Global['getType'] = function(global) {
+    return Module['_BinaryenGlobalGetType'](global);
+  };
+  Global['isMutable'] = function(global) {
+    return Boolean(Module['_BinaryenGlobalIsMutable'](global));
+  };
+  Global['getInitExpr'] = function(global) {
+    return Module['_BinaryenGlobalGetInitExpr'](global);
+  };
+  deriveWrapperInstanceMembers(Global.prototype, Global);
+  Global.prototype['valueOf'] = function() {
+    return this[thisPtr];
+  };
+  return Global;
+})();
+
+// FunctionImport wrapper
+Module['FunctionImport'] = (() => {
+  // Closure compiler doesn't allow multiple `FunctionImport`s at top-level, so:
```

Comment:
As well: The IIFE here is not necessary. Only affects JS built-ins like `Function`.